sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -101,6 +101,128 @@ def hpat_pandas_df_index_impl(df):
+    func_text = []
+    column_list = []
+    column_len = len(df.columns)
+    func_text.append(f'row_len = len(get_dataframe_data(df, 0))')
+
+    for index, column_name in enumerate(df.columns):
+        func_text.append(f'df_col_{column_name} = get_dataframe_data(df, {index})')
+        column_list.append(f'df_col_{column_name}')
+
+    func_text.append(f'df_values = numpy.empty(row_len*{column_len}, numpy.dtype("{numba_common_dtype}"))')
+    func_text.append('for i in range(row_len):')
+    for j in range(column_len):
+        func_text.append(indent + f'df_values[i * {column_len} + {j}] = {column_list[j]}[i]')
+
+    func_text.append(f"return df_values.reshape(row_len, {column_len})\n")
+    func_definition.extend([indent + func_line for func_line in func_text])
+    func_def = '\n'.join(func_definition)
+
+    global_vars = {'pandas': pandas, 'numpy': numpy,
+                   'get_dataframe_data': sdc.hiframes.pd_dataframe_ext.get_dataframe_data}
+
+    return func_def, global_vars
+
+
+@sdc_overload_attribute(DataFrameType, 'values')
+def hpat_pandas_dataframe_values(df):
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+    Pandas API: pandas.DataFrame.values
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/dataframe_values.py
+      :language: python
+      :lines: 27-
+      :caption: The values data of the DataFrame.
+      :name: ex_dataframe_values
+
+    .. command-output:: python ./dataframe/dataframe_values.py
+       :cwd: ../../../examples
+
+    .. seealso::
+
+        :ref:`DataFrame.to_numpy <pandas.DataFrame.to_numpy>`
+            Recommended alternative to this method.
+        :ref:`DataFrame.index <pandas.DataFrame.index>`
+            Retrieve the index labels.
+        :ref:`DataFrame.columns <pandas.DataFrame.columns>`
+            Retrieving the column names.
+
+    .. note::
+
+        The dtype will be a lower-common-denominator dtype (implicit upcasting);
+        that is to say if the dtypes (even of numeric types) are mixed, the one that accommodates all will be chosen.
+        Use this with care if you are not dealing with the blocks.
+        e.g. If the dtypes are float16 and float32, dtype will be upcast to float32. If dtypes are int32 and uint8,
+        dtype will be upcast to int32. By numpy.find_common_type() convention,
+        mixing int64 and uint64 will result in a float64 dtype.
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas DataFrame attribute :attr:`pandas.DataFrame.values` implementation.
+    .. only:: developer
+    Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_values*
+    Parameters
+    -----------
+    df: :obj:`pandas.DataFrame`
+       input arg
+    Returns
+    -------
+    :obj: `numpy.ndarray`
+       return a Numpy representation of the DataFrame
+    """
```

Comment:
Does pandas support Strings? If so, could you please add limitation section, that only numbers are supported?

Reply:
Yes, Pandas supports Strings and mixed values types